### PR TITLE
Made arrow tips rotate up/down

### PR DESCRIPTION
**Pull request by icedcoffeeee**
## Overview: What does this pull request change?

Arrow tips will rotate according to the difference in the z-coordinates

## Motivation and Explanation: Why and how do your changes improve the library?

I noticed vector tips fail to orient correctly when the start and end coords differ in z-coordinates. I added the necessary rotations using their spherical coordinates.

Code used:
```py
class Test(ThreeDScene):
    def construct(self):
        self.set_camera_orientation(PI / 3, -PI / 4)
        self.add(ThreeDAxes())
        self.add(Vector([1, 1, 1]))
```

Before:
Test_ManimCE_v0 9 0

After:
Test2_ManimCE_v0 9 0

## Reviewer Checklist
- [x] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/mobject/geometry.py [hunk 1]
```diff
@@ -177,7 +178,10 @@ def position_tip(self, tip, at_start=False):
         else:
             handle = self.get_last_handle()
             anchor = self.get_end()
-        tip.rotate(angle_of_vector(handle - anchor) - PI - tip.tip_angle)
+        angles = cartesian_to_spherical(handle - anchor)
```
Comment (hydrobeam) [inline manim/mobject/geometry.py:181]: It'd help with clarity to add a comment explaining the mathematical reasoning here.

**Review by hydrobeam — APPROVED** [review]
LGTM! It did feel a little buggy when trying to use `Arrow` in three dimensions and `Arrow3D` does tend to be ridiculously slow, so this is a welcome change.

I've left a comment to add some more explanation to what's going on to rotate the tip

Also might be useful to add some documentation (especially for the return values) to `cartesian_to_spherical`, but that's out of the scope of this PR so no worries if not 😅.

File: manim/mobject/geometry.py (hunk 1 above)
Comment (icedcoffeeee) [reply manim/mobject/geometry.py:181]: Is something like this okay?
```py
tip.rotate(angles[2] - PI - tip.tip_angle) # Rotates the tip along the azimuthal
axis = [np.sin(angles[2]), -np.cos(angles[2]), 0] # Obtains the perpendicular of the tip
tip.rotate(-angles[1] + PI / 2, axis=axis) # Rotates the tip along the vertical wrt the axis
```

Comment (icedcoffeeee) [reply manim/mobject/geometry.py:181]: Also, in my pr #1798 introducing the Plane Mobject, i added the docs for the cartesian_to_spherical function. You might want to take a look at that too.

Comment (hydrobeam) [reply manim/mobject/geometry.py:181]: Yeah that seems solid 👍.

> Also, in my pr #1798 introducing the Plane Mobject, i added the docs for the cartesian_to_spherical function

Ah I must've missed that 😅 I'll take another look at that PR then.

**Review by Darylgolden — CHANGES_REQUESTED** [review]
Minor thing regarding `np.round`.

**Review by behackl — CHANGES_REQUESTED** [review]
The implementation looks good to me, thank you.

If I am not mistaken, this is a duplicate of #972, where the suggested change also fixed the problem, but introduced a minor inconsistency across operating systems.

Could you please add a graphical unit test? In case we can't get a test to run for rotated 3D arrowtips, we should look into increasing the testing tolerance for this test.

**Comment by icedcoffeeee** [comment]
> If I am not mistaken, this is a duplicate of #972, 

Oh I didn't realize it was attempted before 😅 

> Could you please add a graphical unit test?

Yup, done 😄

**Review by behackl — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**